src/App.ts

```diff
@@ -80,6 +80,7 @@ export interface AppOptions {
   clientOptions?: Pick<WebClientOptions, 'slackApiUrl'>;
   socketMode?: boolean;
   developerMode?: boolean;
+  tokenVerificationEnabled?: boolean;
```

Comment:
In addition to the essential changes for #718, I organized the internals of App a bit and added more comments. If other maintainers prefer having these changes in a different pull request, I will move these changes to another pull request.